astroquery/mast/observations.py

```diff
@@ -0,0 +1,902 @@
+
+        super(ObservationsClass, self).__init__()
+
+        self._portal_api_connection = PortalAPI(self._session)
+
+        if mast_token:
+            self._authenticated = self._auth_obj = MastAuth(self._session, mast_token)
+        else:
+            self._auth_obj = MastAuth(self._session)
+
+        self._cloud_connection = None
+
+
+    def _login(self, token=None, store_token=False, reenter_token=False):
+        """
+        Log into the MAST portal.
+
+        Parameters
+        ----------
+        token : string, optional
+            Default is None.
+            The token to authenticate the user.
+            This can be generated at
+            https://auth.mast.stsci.edu/token?suggested_name=Astroquery&suggested_scope=mast:exclusive_access.
+            If not supplied, it will be prompted for if not in the keyring or set via $MAST_API_TOKEN
+        store_token : bool, optional
+            Default False.
+            If true, MAST token will be stored securely in your keyring.
+        reenter_token :  bool, optional
+            Default False.
+            Asks for the token even if it is already stored in the keyring or $MAST_API_TOKEN environment variable.
+            This is the way to overwrite an already stored password on the keyring.
+        """
+
+        return self._auth_obj.login(token, store_token, reenter_token)
+
+    @deprecated(since="v0.3.9", message=("The get_token function is deprecated, "
+                                         "session token is now the token used for login."))
+    def get_token(self):
+        return None
+
+    def session_info(self, silent=None, verbose=None):
+        """
+        Displays information about current MAST user, and returns user info dictionary.
+
+        Parameters
+        ----------
+        silent :
+            Deprecated. Use verbose instead.
+        verbose : bool, optional
+            Default True. Set to False to suppress output to stdout.
+
+        Returns
+        -------
+        response : dict
+        """
+
+        # Dealing with deprecated argument
+        if (silent is not None) and (verbose is not None):
+            warnings.warn(("Argument 'silent' has been deprecated, "
+                           "will be ignored in favor of 'verbose'"), AstropyDeprecationWarning)
+        elif silent is not None:
+            warnings.warn(("Argument 'silent' has been deprecated, "
+                           "and will be removed in the future. "
+                           " Use 'verbose' instead."), AstropyDeprecationWarning)
+            verbose = not silent
+        elif (silent is None) and (verbose is None):
+            verbose = True
+
+        return self._auth_obj.session_info(verbose)
+
+    def logout(self):
+        """
+        Log out of current MAST session.
+        """
+        self._auth_obj.logout()
+        self._authenticated = False
+
+    def _parse_result(self, responses, verbose=False):
```

Comment:
For `_parse_result()` both here and in `MastClass`, is it called anywhere and/or is it needed?

Reply:
It's part of the @async_to_sync functionality. When you mark a function as async the decorator causes the class to make a new function that isn't async which is the output of the async function run through _parse_result.

Reply:
OK, that makes sense.  The fact that that wasn't obvious to me is a good example of why I'm not enamored with decorators.  But I have to admit that this seems like a good case for one since that pattern is repeated over and over again.